### PR TITLE
linux: do not set stack size hard limit

### PR DESCRIPTION
**Pull request by mikdusan**
At main startup, if the ELF auxiliary vector contains a stacksize value, use it as a hint for the minimum stacksize required by the executable.

1. Never lower the hard-limit. Once a hard-limit is lowered, then it can never be increased (including child processes).

2. If hint exceeds hard-limit then clamp hint to hard-limit.

3. If soft-limit exceeds hint then do nothing.

### CODE REVIEW

**Comment by matu3ba** [comment]
Sounds like a good idea.

As I understand it, there is a cli option missing to let the user limit the stack size of a run step (only for compile steps its used) which is relevant for limited systems, ie CI systems in container environments to rule out any subprocess exceeding stack limits.

I did also look into what windows offers and [I found nothing to limit the stack size](https://stackoverflow.com/questions/156510/increase-stack-size-on-windows-gcc)

Do you feel like this is relevant or are the environment messages conclusive enough?

**Review by andrewrk — CHANGES_REQUESTED** [review]
1. Can you try to do it with only 1 syscall? Does it succeed if you pass "unlimited" as max?
2. Please avoid unnecessary use of `var`. Both of these should be `const`.

**Comment by mikdusan** [comment]
> * Can you try to do it with only 1 syscall? Does it succeed if you pass "unlimited" as max?

TL;DR: nope. I cannot see a way that we can set *only* `.cur` and leave `.max` untouched without first reading it.

`RLIM_INFINITY` is defined as `0xffffffffffffffff` (`maxInt(u64)`) and unfortunately, it has no symbolic meaning. Thus, if I try to set `.max = RLIM_INFINITY` it only works if that is the same value as *already* active. For example, if active .max is 16 MiB, then bumping +1 to `16777217` yields `error.PermissionDenied` because never allowed to increase hard-limit. And `0` has no special meaning.

> * Please avoid unnecessary use of `var`. Both of these should be `const`.

Done. (my initial thoughts were that `2 * var` was better than `2 * const + 1 * tmp` but hey things aren't so simple).

**Comment by andrewrk** [comment]
Thanks!
